components/reddit/actions/list-subreddits/list-subreddits.mjs

```diff
@@ -0,0 +1,107 @@
+import reddit from "../../reddit.app.mjs";
+import get from "lodash/get.js";
+import { axios } from "@pipedream/platform";
+
+export default {
+  type: "action",
+  key: "reddit-action-list-subreddits",
+  version: "0.0.1",
+  name: "List Subreddits",
+  description: "List subreddits based in a search criteria. [See the docs here](https://www.reddit.com/dev/api/#GET_subreddits_search)",
+  props: {
+    reddit,
+    subReddit: {
+      propDefinition: [
+        reddit,
+        "subreddit",
+      ],
+    },
+    limit: {
+      propDefinition: [
+        reddit,
+        "limit",
+      ],
+    },
+    before: {
+      propDefinition: [
+        reddit,
+        "before",
+      ],
+    },
+    after: {
+      propDefinition: [
+        reddit,
+        "after",
+      ],
+    },
+    count: {
+      propDefinition: [
+        reddit,
+        "count",
+      ],
+    },
+    sort: {
+      type: "string",
+      label: "Sort",
+      description: "One of: `relevance`, `activity`",
+      options: [
+        "relevance",
+        "activity",
+      ],
+      optional: true,
+    },
+    showUsers: {
+      type: "boolean",
+      label: "Show Users",
+      description: "A boolean value",
+      optional: true,
+    },
+    srDetails: {
+      type: "boolean",
+      label: "Subreddit Details",
+      description: "Expand Subreddits",
+      optional: true,
+    },
+    typeaheadActive: {
+      type: "boolean",
+      label: "Typeahead Active",
+      description: "Boolean value or None",
+      optional: true,
+    },
+    show: {
+      type: "string",
+      label: "Show",
+      description: "The string `all`",
+      optional: true,
+    },
+    searchQueryId: {
+      type: "string",
+      label: "Search Queue ID",
+      description: "An `uuid`",
+      optional: true,
+    },
+  },
+  async run({ $ }) {
+    const params = {
+      q: get(this.subReddit, "value", this.subReddit),
```

Comment:
The `q` parameter seems to be used to search the title _and_ description of subreddits according to the [API docs](https://www.reddit.com/dev/api/#GET_subreddits_search). Using the `subReddit` prop as the value of `q` suggests only subreddit _titles_ are searched.
Do you think we could replace the `subReddit` prop in this action with a `q` prop instead?